lib/trax/model/struct.rb

```diff
@@ -11,86 +11,67 @@ class Struct < ::Hashie::Dash
       include ::Hashie::Extensions::Dash::PropertyTranslation
       include ::ActiveModel::Validations
 
-      class_attribute :fields
-
-      def self.inherited(subklass)
-        super(subklass)
-      end
+      # note that we must explicitly set default or blank values for all properties.
+      # It defeats the whole purpose of being a 'struct'
+      # if we fail to do so, and it makes our data far more error prone
+      DEFAULT_VALUES_FOR_PROPERTY_TYPES = {
+        :boolean_property => nil,
+        :string_property  => "",
+        :struct_property  => {},
+        :enum_property    => nil,
+      }.with_indifferent_access.freeze
 
       def self.fields_module
         @fields_module ||= begin
-          const_set("Fields", ::Module.new)
-          const_get("Fields").extend(::Trax::Model::Attributes::Fields)
+          module_name = "#{self.name}::Fields"
+          ::Trax::Core::NamedModule.new(module_name, ::Trax::Model::Attributes::Fields)
         end
       end
 
+      def self.fields
+        fields_module
+      end
+
       def self.boolean_property(name, *args, **options, &block)
         name = name.is_a?(Symbol) ? name.to_s : name
-        klass = fields_module.const_set(name.camelize, ::Class.new(::Trax::Model::Attributes[:boolean]::Attribute))
-        options[:default] = options.key?(:default) ? options[:default] : nil
+        klass_name = "#{fields_module.name.underscore}/#{name}".camelize
```

Comment:
ya that was the idea behind some of the changes here, started making interface more uniform. -- The main thing I want to allow for in future is Trax::Core to have a struct, which has the behavior of this without the search scope stuff, then trax model would just extend the instances with the search definition/validation behavior and what not. Doing so would allow for separation of DDL from model, so that we could share the domain objects in different classes, i.e. a locale settings struct that gets defined once and included by many models.